pytest_plugins/auto_vault.py

```diff
@@ -0,0 +1,19 @@
+"""Plugin enables pytest to notify and update the requirements"""
+import os
+import re
+import subprocess
+from pathlib import Path
+
+
+def pytest_addoption(parser):
+    """Options to allow user to update the requirements"""
+    envpath = Path('.env')
+    # check if this is being executed by a user and not automation/CI
+    if (
+        re.search(r'\s*#.*VAULT_SECRET_ID_FOR_DYNACONF', envpath.read_text())
+        and 'VAULT_SECRET_ID_FOR_DYNACONF' not in os.environ
+    ):
+        vstatus = subprocess.run("make vault-status", shell=True, capture_output=True).returncode
+        if vstatus != 0:
+            print('Vault token is expired, Logging you in ...')
+            subprocess.run('make vault-login', shell=True, stdout=subprocess.PIPE)
```

Comment:
Could you use `timeout` parameter here to make it safer?

Reply:
I can add but this is step you have to authenticate if you are not logged into vault, you should not continue.